packages/runner/src/context.ts

```diff
@@ -52,6 +51,9 @@ export function withTimeout<T extends (...args: any[]) => any>(
         // `unref` might not exist in browser
         timer.unref?.()
       }),
+      Promise.resolve(fn(...args)).then((result) => {
+        return new Promise(resolve => setTimeout(resolve, 0, result))
+      }),
```

Comment:
I took a while to understand this but the idea is more or less like this? Executing `fn` in sync as usual, but holding off returning until `setTimeout(.., 0)` to flush other `setTimeout` callbacks. I think code is fine as is, but it's just my commentary.

```ts
      (async () => {
        const result = await fn();
        await new Promise(resolve => setTimeout(resolve, 0))
        return result;
      })()
```

Reply:
Thanks a lot! Yes that's the same thing, I can try refactor with your code if it's best.